src/nvim/ex_cmds.c

```diff
@@ -2946,6 +2953,20 @@ void do_sub(exarg_T *eap)
   int start_nsubs;
   int save_ma = 0;
 
+  //  list to save matched lines
+  //  Note: Could not use nolint with rule, line would be > 100 chars
+  klist_t(matchedline_T) *lmatch = kl_init(matchedline_T); // NOLINT
+
+  sub_done = 0;
+  bool last_is_slash = false;
+  char_u *c = eap->arg;
+
+  // checking if the last char of the cmdline is a '/'
+  while (*c != '\0') {
+    c++;
+  }
+  last_is_slash = (*(c-1) == '/');
```

Comment:
Ok, following the original `do_sub` I just left `last_is_slash` set to `false` in that case. Seems to work.